app/src/main/AndroidManifest.xml

```diff
@@ -312,6 +312,10 @@
             android:name=".feature.genericauth.GenericAuthActivity"
             android:windowSoftInputMode="adjustResize" />
 
+        <activity
```

Comment:
Yeah maybe we should look at migrating to compose navigation

Reply:
Compose navigation is heavily tied with Jetpack Navigation which I am not sure if it can work as a thing in the "leaf" nodes. I've only seen it encompass the entire application navigation graph. But might be worth looking into.

Reply:
It's not impossible to use it for this use-case.

Note that implementing navigation on your own with Compose is completely trival, so the investment need not be that big 😄

Here's an example:

```kotlin
sealed class NavigationDestination {
    object Example1: NavigationDestination()
    object Example2: NavigationDestination()
}

@Composable
fun Navigation() {
    var currentDestination by remember { mutableStateOf(NavigationDestination.Example1) }
    val navigateTo = { navigationDestination: NavigationDestination -> currentDestination = navigationDestination }
    when (currentDestination) {
        NavigationDestination.Example1 -> { Example1Composable(navigateTo) }
        NavigationDestination.Example2 -> { Example2Composable(navigateTo) }
    }
}
```

Reply:
caveat: I didn't compile this code, I simply wrote it inline in this comment.

Navigation in Compose is actually that simple 😅

Reply:
Oh if we mean like this then absolutely it's trivial. Could even slap a CrossFade composable around this and get animations too.
But I don't know how well this handles backstacks, restoration after process death, deep links to the screens we'll be making and much more I'm missing probably. Depending on how much we do this the more things we'd need to consider. And the more we might do it the more we'd need to consider a proper investment in androidx-navigation.

Reply:
That's true Stelios. I suggest we take a look at navigation together as a separate PR.